exist-core/src/test/java/org/exist/xquery/WindowClauseTest.java

```diff
@@ -0,0 +1,474 @@
+        }
+
+        assertTrue(expr.getFirst() instanceof WindowExpr, "Expression should be of type WindowExpr");
+        assertEquals("first", ((WindowExpr) expr.getFirst()).getWindowStartCondition().getCurrentItem().getStringValue());
+        assertEquals(null, ((WindowExpr) expr.getFirst()).getWindowStartCondition().getPosVar());
+        assertEquals(null, ((WindowExpr) expr.getFirst()).getWindowStartCondition().getPreviousItem());
+        assertEquals(null, ((WindowExpr) expr.getFirst()).getWindowStartCondition().getNextItem());
+        assertEquals(null, ((WindowExpr) expr.getFirst()).getWindowEndCondition());
+    }
+
+    @Test
+    public void slidingWindowAvgReturn() throws RecognitionException, XPathException, TokenStreamException, QName.IllegalQNameException {
+        final String query = "xquery version \"3.1\";\n" +
+                "for sliding window $w in (2, 4, 6, 8, 10, 12, 14)\n" +
+                "    start at $s when fn:true()\n" +
+                "    only end at $e when $e - $s eq 2\n" +
+                "return avg($w)";
+
+        // parse the query into the internal syntax tree
+        final XQueryContext context = new XQueryContext();
+        context.importModule(Function.BUILTIN_FUNCTION_NS, "fn", new AnyURIValue[] { new AnyURIValue("java:" + FnModule.class.getName()) });
+        final XQueryLexer lexer = new XQueryLexer(context, new StringReader(query));
+        final XQueryParser xparser = new XQueryParser(lexer);
+        xparser.xpath();
+        if (xparser.foundErrors()) {
+            fail(xparser.getErrorMessage());
+        }
+
+        final XQueryAST ast = (XQueryAST) xparser.getAST();
+
+        final XQueryTreeParser treeParser = new XQueryTreeParser(context);
+        final PathExpr expr = new PathExpr(context);
+        treeParser.xpath(ast, expr);
+        if (treeParser.foundErrors()) {
+            fail(treeParser.getErrorMessage());
+        }
+
+        assertTrue(expr.getFirst() instanceof WindowExpr, "Expression should be of type WindowExpr");
+        assertEquals(WindowExpr.WindowType.SLIDING_WINDOW, ((WindowExpr) expr.getFirst()).getWindowType());
+        assertEquals(null, ((WindowExpr) expr.getFirst()).getWindowStartCondition().getCurrentItem());
+        assertEquals(new QName("s"), ((WindowExpr) expr.getFirst()).getWindowStartCondition().getPosVar());
+        assertEquals(null, ((WindowExpr) expr.getFirst()).getWindowStartCondition().getPreviousItem());
+        assertEquals(null, ((WindowExpr) expr.getFirst()).getWindowStartCondition().getNextItem());
+        assertEquals(new QName("e"), ((WindowExpr) expr.getFirst()).getWindowEndCondition().getPosVar());
+        assertEquals(null, ((WindowExpr) expr.getFirst()).getWindowEndCondition().getPreviousItem());
+        assertEquals(null, ((WindowExpr) expr.getFirst()).getWindowEndCondition().getNextItem());
+        assertEquals(true, ((WindowExpr) expr.getFirst()).getWindowEndCondition().isOnly());
```

Comment:
Any special reason not using `assertTrue` rsp. `assertNull` for the boolean and null cases?